api/src/routes/settings.test.ts

```diff
@@ -186,6 +186,158 @@ describe('settingRoutes', () => {
       });
     });
 
+    describe('/update-my-username', () => {
+      test('PUT returns an error when the username uses special characters', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'twaha@'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'Username twaha@ contains invalid characters',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is an endpoint', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'german'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-taken',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is a bad word', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'ass'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'flash.username-taken',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is a https status code', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: '404'
+        });
+
+        expect(response?.statusCode).toEqual(400);
+
+        expect(response?.body).toEqual({
+          message: 'Username 404 is a reserved error code',
+          type: 'info'
+        });
+      });
+
+      test('PUT returns an error when the username is shorter than 3 characters', async () => {
+        const response = await superRequest('/update-my-username', {
+          method: 'PUT',
+          setCookies
+        }).send({
+          username: 'fo'
+        });
+
+        expect(response?.statusCode).toEqual(500);
```

Comment:
```suggestion
        expect(response?.statusCode).toEqual(400);
```
This should be 400, since it's not a problem with the server - it's a problem with the request.

While there might be exceptions to this rule, 5xx errors are for when something unexpected happened on the server. This is just bad input.

Since this is generated by the validation, you can get access to `req.validationError` by using setting the `attachValidation` option to true: https://www.fastify.io/docs/latest/Reference/Validation-and-Serialization/#error-handling

That allows you to send whatever response you like, instead of the default one that Fastify supplies.